### ISSUE TITLE
GraphQL `ProductFilterInput` Requires Field to be a String

### ISSUE THREAD

**Issue by maxbucknell** [magento/magento2#14312]
Every field of the `ProductFilterInput` input field in `Magento_CatagolGraphQl::etc/graphql.xml` has the type `FilterTypeInput`. The problem here is that `FilterTypeInput` requires most of its arguments to be strings.

For example, I might want to select products that do not have a special price set. This might be some query like:

```graphql
{
  products(filter: { special_price: { null: true } }) {
    total_count
  }
}
```

The devdocs, and basic intuition seem to suggest this is okay. However, this doesn't work because `Magento_GraphQl::etc/graphql.xml` requires the `null` field in `FilterTypeInput` to be a string.

### Preconditions

1. This is GraphQL, so it's in `2.3-develop`, natch.

### Steps to reproduce

1. Try to submit the query written above.

### Expected result

Based on the documentation, I would expect this to be a legal query.

Receive a response like this:

```json
{
  "data": {
    "products": {
      "total_count": 0
    }
  }
}
```

### Actual result

Response:

```json
{
  "errors": [
    {
      "message": "Argument \"filter\" has invalid value {special_price: {null: true}}.\nIn field \"special_price\": In field \"null\": Expected type \"String\", found true.",
      "category": "graphql",
      "locations": [
        {
          "line": 2,
          "column": 20
        }
      ]
    }
  ]
}
```

This also applies to other fields. For example, I was unable to filter products by category ids.

**Comment by maxbucknell** [comment]
After some research, I have found that this isn't what is causing this particular issue. While there is an issue with using the same `FilterTypeInput` field and requiring all fields to be a `String`, this particular issue is caused by magento/magento2/issues/14355, and I will fix that.

I believe that the `FilterTypeInput` issue does require further discussion, though.

**Comment by misha-kotov** [comment]
Hey @maxbucknell since the root cause is described in the other ticket, can we close this issue? We can discuss `FilterTypeInput` in slack or offline

**Comment by magento-engcom-team** [comment]
Hi @maxbucknell. Thank you for your report. 
The issue has been fixed in magento/magento2#14356 by @maxbucknell in 2.3-develop branch
Related commit(s):
 - [7a3ea4ef23584d3dfdd1f414dad69858c3c46ac9](https://github.com/magento/magento2/commit/7a3ea4ef23584d3dfdd1f414dad69858c3c46ac9)
 - [4589cb0ad644290e5e8d968b275bc5e2357bb349](https://github.com/magento/magento2/commit/4589cb0ad644290e5e8d968b275bc5e2357bb349)

The fix will be available with the upcoming 2.3.0 release.
